core/state/statedb.go

```diff
@@ -468,6 +466,10 @@ func (s *StateDB) updateStateObject(obj *stateObject) {
 
 	// If state snapshotting is active, cache the data til commit
 	if s.snap != nil {
+		// If the account is an empty resurrection, unmark the storage nil-ness
+		if storage, ok := s.snapStorage[obj.addrHash]; storage == nil && ok {
```

Comment:
I'm not convinced this is correct. There are a couple of things that can happen: 

- Old code (before `CREATE2`): A contract with storage is selfdestructed in tx `n`. In tx `n+1`, someone sends a wei to the address, and the account is recreated. The desired end-state should be that the storage has become `nil` and the account exists. 

- New code, withe CREATE2: A contract is killed in tx `n`. In tx `n+1`, the contract is recreated, and the initcode sets new storaege slots. So the old storage slots are all cleared, and there are now new storage slots set. We need to handle this (we don't currently)